components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

```diff
@@ -42,18 +42,13 @@ fun String.isGeoLocation() = re.geoish.matches(this)
  */
 fun String.toDate(format: String, locale: Locale = Locale.ROOT): Date {
     val formatter = SimpleDateFormat(format, locale)
-    return if (!this.isEmpty()) {
+    return if (isNotEmpty()) {
         formatter.parse(this)
     } else {
         Date()
     }
 }
 
-/**
- * Converts a [String] to a [Uri] object.
- */
-fun String.toUri() = Uri.parse(this)
```

Comment:
The A-C extension function was removed

Reply:
Oops thanks, I didn't catch that!

Reply:
Let's deprecate first and allow for less API breakage. Whistler would be a good time to discuss a process around deprecating and removing functionality. It's something that have come up more often recently, but where we do not have any process guidelines.

Reply:
I was told by @csadilek that deprecating also causes API breakage because lint will fail. Should I hold off on this refactoring until another time?

Reply:
Oh yeah, lint is super annoying. In other projects with strict rules failing lint is fine; however it breaks our own code and tests if we still use the old method anywhere, but I think we do not since you migrated that?

Reply:
Lint suppression is much easier than applying changes to code (I need to find what I should use instead and apply it around the project - it can be extremely painful).

Luckily, Kotlin has deprecation on steroids:

```kotlin
@Deprecated("Use `newImplementation()` instead", ReplaceWith("newImplementation(args)", "mozilla.x"))
fun oldImplementationWith(args) {
}
```

that will allow clients to change code in the whole project with two clicks. :)